src/main.rs

```diff
@@ -111,8 +113,19 @@ fn setup_pg_pool(api_config: &ApiConfig) -> deadpool_postgres::Pool {
 
 #[actix_web::main]
 async fn main() -> Result<(), std::io::Error> {
-    let Ok(config) = confy::load_path("tsom_api_config.toml") else {
-        panic!("failed to load config, please check tsom_api_config.toml");
+    let config = match confy::load_path("tsom_api_config.toml") {
+        Ok(config) => config,
+        Err(ConfyError::BadTomlData(err)) => panic!(
+            "an error occured on the parsing of the file tsom_api_config.toml:\n{}",
+            err.message()
+        ),
+        Err(ConfyError::GeneralLoadError(err)) => panic!(
+            "an error occured on the loading of the file tsom_api_config.toml:\n{}",
+            err
```

Comment:
this can leak the toml data, which is why I didn't put it.

Reply:
the leak of the file content is with `ConfyError::BadTomlData(err)` and to avoid that I yes `err.message()` to get the pure message without the content of it but I can add `.message()` at the line 124